test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,445 @@
+                        EXTANT_AGREEMENT_ID,
+                        Units.ether(1),
+                        mockToken.address,
+                        { from: PAYER }
+                    );
+                });
+
+                it("returns 2 ether", async () => {
+                    await expect(termsContract.getValueRepaidToDate.callAsync(
+                      EXTANT_AGREEMENT_ID
+                    )).to.eventually.bignumber.equal(Units.ether(2));
+                });
+            });
+        });
+    });
+
+    describe("#getExpectedRepaymentValue", () => {
+
+        describe("when termsContract associated w/ debt agreement is not `this`", () => {
+            it("should throw");
+        });
+
+        describe("when termsContractParameters associated w/ debt agreement malformed", () => {
+
+          const principalPlusInterest = Units.ether(10);
+          const amortizationUnitType = new BigNumber(10); // invalid unit code.
+          const termLength = new BigNumber(10);
+
+          const invalidTermsParams = hexifyParams(principalPlusInterest, amortizationUnitType, termLength);
+
+            describe("amortizationUnitType is not one of the valid types", () => {
+
+                before(async () => {
+                    await mockRegistry.mockGetTermsContractParameters.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        invalidTermsParams
+                    );
+                });
+
+                it("should throw", async () => {
+                    await expect(termsContract.getExpectedRepaymentValue.callAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        new BigNumber(moment().unix())
+                    )).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+            });
+        });
+
+        describe("when termsContractParameters associated w/ debt agreement are well-formed", () => {
+
+            /*
+            The params define a simple interest contract with the below valid terms:
+              - Principal plus interest: 12 ether
+              - Amortization Unit Type: year
+              - Term length: 3 years
+            */
+            const principalPlusInterest = Units.ether(12);
+            const amortizationUnitType = new BigNumber(4); // unit code for years.
+            const termLength = new BigNumber(3); // term is three years.
+
+            const validTermsParams = hexifyParams(principalPlusInterest, amortizationUnitType, termLength);
+
+            const ORIGIN_MOMENT = moment();
+            const BLOCK_ISSUANCE_TIMESTAMP = ORIGIN_MOMENT.unix();
+
+            const ZERO_AMOUNT = Units.ether(0);
+            const INSTALLMENT_AMOUNT = principalPlusInterest.div(termLength);
+            const FULL_AMOUNT = principalPlusInterest;
+
+            it("unpacks valid params", async () => {
+                var params = await termsContract.unpackParametersFromBytes.callAsync(
```

Comment:
Might have flagged this in another PR so apologies if this is a duplicate comment, but `unpackParametersFromBytes` is a public method that should be tested under a different scope (right now, it is in `#getExpectedRepaymentValue`)